docs/contributing/contributing-code/contributing-code-debugging/radius-os-processes-debugging.md

```diff
@@ -0,0 +1,480 @@
+- Database setup verification
+- Management scripts (start/stop/status)
+- Incremental builds for individual components
+- Convenient `./drad` symlink in workspace root for easy CLI access
+- Debug CLI wrapper `./drad` with automatic UCP endpoint configuration
+
+## Prerequisites
+
+The automation checks for all required tools. Install any missing prerequisites:
+
+### Required Tools
+- **Go 1.21+** - `go version`
+- **Delve debugger** - `dlv version` (Go debugger for VS Code integration)
+- **kubectl** - Kubernetes cluster access
+- **psql** - PostgreSQL client for database verification
+- **terraform** - Terraform CLI for recipe execution
+- **docker** - To host k3d
+
+#### Quick PostgreSQL Setup (if you don't already have one)
+
+The automation automatically detects and works with different PostgreSQL setups:
+
+**Option 1: Docker PostgreSQL (Recommended for Development)**
+If you need a throwaway local PostgreSQL instance for debugging Radius:
+
+```bash
+# Quick setup (uses postgres superuser - works with current automation)
+docker run --name radius-postgres \
+   -e POSTGRES_PASSWORD=radius_pass \
+   -p 5432:5432 \
+   -d postgres:15
+```
+
+**Alternative Docker setup with custom user:**
+```bash
+# Custom user setup (more production-like)
+docker run --name dev-postgres \
+  -e POSTGRES_USER=radius_user \
+  -e POSTGRES_PASSWORD=radius_pass \
+  -e POSTGRES_DB=radius \
+  -p 5432:5432 \
+  -d postgres
+```
+*Note: If using the custom user setup, you'll need to update the PostgreSQL connection variables in `build/scripts/start-radius.sh` to match.*
+
+**Option 2: Local PostgreSQL Installation (Homebrew/System)**
+If you already have PostgreSQL installed locally (via Homebrew, apt, etc.), the automation will detect and use it automatically.
+
+**Automatic Database Setup**
+The automation handles all database setup automatically when you run `make debug-start`:
+- Creates required users (`applications_rp`, `ucp`) with proper passwords
+- Creates databases with correct ownership
+- Sets up proper permissions and table structures
+- Works with both Docker and local PostgreSQL installations
+
+The automation will verify connectivity during `make debug-check-prereqs` and create all required users/databases during `make debug-start`. No manual database setup is required.
+
+### Optional Tools
+- **VS Code** - For integrated debugging experience
+
+### Installation Commands
+
+**macOS:**
+```bash
+# Core tools
+brew install go kubectl postgresql
+
+# Delve debugger
+go install github.com/go-delve/delve/cmd/dlv@latest
+
+# Add Go binaries to PATH (required for delve)
+echo 'export PATH="$HOME/go/bin:$PATH"' >> ~/.zshrc
+source ~/.zshrc
+
+# Terraform (HashiCorp official method)
+brew tap hashicorp/tap
+brew install hashicorp/tap/terraform
+
+# Optional tools
+brew install --cask docker
+brew install --cask visual-studio-code
+```
+
+**Ubuntu/Debian:**
+```bash
+# Core tools
+sudo apt update
+sudo apt install golang-go kubectl postgresql-client
```

Comment:
issue: `apt install golang-go` installs distro-specific version of Go (currently 1.22). The project uses 1.25.